Add more @showarg examples

bluesmoon opened the pull request:
Adds a few more examples to show how @showarg works with literals and interpolated string expressions.

bluesmoon committed:
Add more @showarg examples

Adds a few more examples to show how @showarg works with literals and interpolated string expressions.

fingolfin reviewed (approved):
I find the selection of examples a bit arbitrary but if you find it helpful, why not :-)

fingolfin committed:
Merge branch 'master' into patch-4

vtjnash commented:
The docstring REPL does not permit these sort of inline comments. I am not sure this is that useful, since it is also just showing that different expressions parse differently, which is probably explained elsewhere (though I agree that it can take a while to fully appreciate the implications of that)

fingolfin commented on doc/src/manual/metaprogramming.md:
```diff
@@ -629,6 +629,18 @@ julia> @showarg(1+1)
 julia> @showarg(println("Yo!"))
 :(println("Yo!"))
+
+# Numeric literal
+julia> @showarg(1)
```
You'd have to do it like this:
```suggestion
julia> @showarg(1)   # Numeric literal
```

fingolfin commented:
@vtjnash that's a valid argument against merging this, of course.

So who makes the decision whether to reject this, or whether to ask for further improvements? I personally could live with either, but we should not leave @bluesmoon hanging

bluesmoon commented:
I'll update the MR.

bluesmoon committed:
Move comments to code line for docstring

docstring REPL does not like inline comments on a line by itself, so move the comments to the code line (the line with the `julia>` prompt)

bluesmoon commented:
I've updated the MR with @vtjnash's suggestion.
